lightkurve/prf/scenemodels.py

```diff
@@ -0,0 +1,670 @@
+            `scipy.optimize.minimize`.
+
+        Returns
+        -------
+        result : ``SceneParameters`` object
+            Fitted parameters plus fitting diagnostics.
+        """
+        loss = loss_function(data, self, prior=self._logp_prior)
+        fit = loss.fit(x0=self.get_initial_guesses().to_array(), method=method, **kwargs)
+        result = self.params.from_array(fit.x)
+        result.uncertainties = loss.loglikelihood.uncertainties(fit.x)
+        result.predicted_image = self._predict(fit.x)
+        result.residual_image = data - result.predicted_image
+        result.loss_value = fit.fun
+        result.opt_result = fit
+        return result
+
+    def plot(self, *params, **kwargs):
+        """Plots an image of the model for a given point in the parameter space."""
+        img = self.predict(*params)
+        plot_image(img,
+                   title='Scene Model, Channel: {}'.format(self.prfmodel.channel),
+                   extent=(self.prfmodel.column, self.prfmodel.column + self.prfmodel.shape[1],
+                           self.prfmodel.row, self.prfmodel.row + self.prfmodel.shape[0]),
+                   **kwargs)
+
+    def diagnostics(self, data, *params, **kwargs):
+        """Plots an image of the model for a given point in the parameter space."""
+        fit = self.fit(data)
+        plot_image(data,
+                   title='Observed Data, Channel: {}'.format(self.prfmodel.channel),
+                   extent=(self.prfmodel.column, self.prfmodel.column + self.prfmodel.shape[1],
+                           self.prfmodel.row, self.prfmodel.row + self.prfmodel.shape[0]),
+                   **kwargs)
+        plot_image(fit.predicted_image,
+                   title='Predicted Image, Channel: {}'.format(self.prfmodel.channel),
+                   extent=(self.prfmodel.column, self.prfmodel.column + self.prfmodel.shape[1],
+                           self.prfmodel.row, self.prfmodel.row + self.prfmodel.shape[0]),
+                   **kwargs)
+        plot_image(fit.residual_image,
+                   title='Residual Image, Channel: {}'.format(self.prfmodel.channel),
+                   extent=(self.prfmodel.column, self.prfmodel.column + self.prfmodel.shape[1],
+                           self.prfmodel.row, self.prfmodel.row + self.prfmodel.shape[0]),
+                   **kwargs)
+
+
+class PRFPhotometry(object):
+    """This class performs PRF Photometry on TPF-like data given a ``SceneModel``.
+
+    This class exists because a ``SceneModel`` object is designed to fit only
+    one cadence at a time.  This class makes it easy to fit a large number
+    of cadences and obtain the resulting LightCurve.
+
+    Attributes
+    ----------
+    model : instance of SceneModel
+        Model which will be fit to the data
+    """
+    def __init__(self, model):
+        self.model = model
+        self.results = []
+
+    def run(self, tpf_flux, pos_corr1=None, pos_corr2=None):
```

Comment:
I think it's worth adding a parameter ``cadences`` here in case users want to fit only a subset of cadences?

Reply:
Done!